### PR TITLE
[browser][io] Workaround for issue MoveDirectory

### PR DESCRIPTION
**Pull request by kjpou1**
- Issue workaround while waiting for emscripten fix.  https://github.com/dotnet/runtime/issues/40305
- The following code checks for the existence of the source and destination directories which replaces the same code in the emscripten code.

emscripten tracking issue:  https://github.com/emscripten-core/emscripten/issues/11804

This will allow the issue to be resolved in the tests without waiting for emscripten release.

### CODE REVIEW

File: src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs [hunk 1]
```diff
@@ -381,6 +381,17 @@ public static void MoveDirectory(string sourceFullPath, string destFullPath)
                 throw new IOException(SR.Format(SR.IO_AlreadyExists_Name, destFullPath));
             }
 
+#if TARGET_BROWSER
+            if (!Directory.Exists(Path.GetDirectoryName(sourceFullPath)))
```
Comment (akoeplinger) [inline src/libraries/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs:385]: I'd add a comment which explains this is a workaround:
```suggestion
            // renaming a file doesn't return correct error code on emscripten if one of the parent paths does not exist,
            // manually workaround it for now (https://github.com/dotnet/runtime/issues/40305)
            if (!Directory.Exists(Path.GetDirectoryName(sourceFullPath)))
```

**Review by akoeplinger — APPROVED** [review]
Looks good, just maybe add one comment.

**Comment by danmosemsft** [comment]
Did we pick up the fix now?

**Comment by lewing** [comment]
@danmosemsft thanks for the reminder https://github.com/dotnet/runtime/pull/46154

**Pull request merged**